### PR TITLE
Add cost-analysis preview CLI flag

### PR DESCRIPTION
**Pull request by matthchr**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/aks-preview/azext_aks_preview/managed_cluster_decorator.py [hunk 1]
```diff
@@ -2276,6 +2276,60 @@ def get_nodepool_taints(self) -> Union[List[str], None]:
...
+
+    def get_enable_cost_analysis(self) -> bool:
+        """Obtain the value of enable_cost_analysis.
+
+        :return: bool
+        """
+        return self._get_enable_cost_analysis(enable_validation=True)
+
+    def _get_disable_cost_analysis(self, enable_validation: bool = False) -> bool:
+        """Internal function to obtain the value of disable_vpa.
```
Comment (babbageclunk) [inline src/aks-preview/azext_aks_preview/managed_cluster_decorator.py:2306]:
```suggestion
        """Internal function to obtain the value of disable_cost_analysis.
```

File: src/aks-preview/azext_aks_preview/managed_cluster_decorator.py [hunk 2]
```diff
@@ -2276,6 +2276,60 @@ def get_nodepool_taints(self) -> Union[List[str], None]:
...
+        # Read the original value passed by the command.
+        disable_cost_analysis = self.raw_param.get("disable_cost_analysis")
+
+        # This option is not supported in create mode, hence we do not read the property value from the `mc` object.
+        # This parameter does not need dynamic completion.
+        if enable_validation:
+            if disable_cost_analysis and self._get_enable_cost_analysis(enable_validation=False):
+                raise MutuallyExclusiveArgumentError(
+                    "Cannot specify --enable-cost-analysis and --disable-cost-analysis at the same time."
+                )
```
Comment (babbageclunk) [inline src/aks-preview/azext_aks_preview/managed_cluster_decorator.py:2322]: It seems odd that you have this check in both directions - is this needed to handle both orders of the options? (I'm assuming this is a pattern used with other settings as well.)

**Review by babbageclunk — APPROVED** [review]

File: src/aks-preview/azext_aks_preview/managed_cluster_decorator.py (hunk 1 above)
Comment (davefellows) [inline src/aks-preview/azext_aks_preview/managed_cluster_decorator.py:2306]: typo
```suggestion
        """Internal function to obtain the value of disable_cost_analysis
```

File: src/aks-preview/azext_aks_preview/managed_cluster_decorator.py (hunk 2 above)
Comment (matthchr) [reply src/aks-preview/azext_aks_preview/managed_cluster_decorator.py:2322]: I copied this pattern from disable_vpa but agree it is sorta weird we check in both directions. Let me confirm I don't need both and if so I'll remove one.

Comment (matthchr) [reply src/aks-preview/azext_aks_preview/managed_cluster_decorator.py:2322]: yeah only one direction is needed. Removed the other one

File: src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py [hunk 3]
```diff
@@ -7560,6 +7560,78 @@ def test_aks_create_with_enable_network_observability(self, resource_group, reso
         self.cmd(
             'aks delete -g {resource_group} -n {name} --yes --no-wait', checks=[self.is_empty()])
 
+    @AllowLargeResponse()
+    @AKSCustomResourceGroupPreparer(random_name_length=17, name_prefix='clitest', location='westus2',
+                                    preserve_default_location=True)
+    def test_aks_create_with_enable_cost_analysis(self, resource_group, resource_group_location):
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:7566]: Queued [live test](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=78890009&view=results) to validate the change.

File: src/aks-preview/azext_aks_preview/_help.py [hunk 4]
```diff
@@ -570,6 +570,9 @@
         - name: --nodepool-taints
           type: string
           short-summary: The node taints for all node pools in this cluster.
+        - name: --enable-cost-analysis
+          type: bool
+          short-summary: Enables exporting Kubernetes Namespace and Deployment details to the Cost Analysis views in the Azure portal. For more information see aka.ms/aks/docs/cost-analysis.
```
Comment (zhoxing-ms) [inline src/aks-preview/azext_aks_preview/_help.py:575]:
```suggestion
          short-summary: Enable exporting Kubernetes Namespace and Deployment details to the Cost Analysis views in the Azure portal. For more information see aka.ms/aks/docs/cost-analysis.
```

File: src/aks-preview/azext_aks_preview/_help.py [hunk 5]
```diff
@@ -1066,6 +1069,12 @@
         - name: --enable-network-observability
           type: bool
           short-summary: Enable network observability on a cluster.
+        - name: --enable-cost-analysis
+          type: bool
+          short-summary: Enables exporting Kubernetes Namespace and Deployment details to the Cost Analysis views in the Azure portal. For more information see aka.ms/aks/docs/cost-analysis.
+        - name: --disable-cost-analysis
+          type: bool
+          short-summary: Disables exporting Kubernetes Namespace and Deployment details to the Cost Analysis views in the Azure portal.
```
Comment (zhoxing-ms) [inline src/aks-preview/azext_aks_preview/_help.py:1077]:
```suggestion
          short-summary: Enable exporting Kubernetes Namespace and Deployment details to the Cost Analysis views in the Azure portal. For more information see aka.ms/aks/docs/cost-analysis.
        - name: --disable-cost-analysis
          type: bool
          short-summary: Disable exporting Kubernetes Namespace and Deployment details to the Cost Analysis views in the Azure portal.
```

File: src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py (hunk 3 above)
Comment (FumingZhang) [reply src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:7566]: Requeued [new one](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=78954450&view=results)

File: src/aks-preview/azext_aks_preview/_help.py (hunk 4 above)
Comment (matthchr) [reply src/aks-preview/azext_aks_preview/_help.py:575]: Done

File: src/aks-preview/azext_aks_preview/_help.py (hunk 5 above)
Comment (matthchr) [reply src/aks-preview/azext_aks_preview/_help.py]: Done

**Comment by matthchr** [comment]
@zhoxing-ms What else needs to be done on this PR before it merges? If nothing, can we merge?

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**